todo/src/util/defineClass.js

```diff
@@ -0,0 +1,30 @@
+export default (function() {
+  const defineClass = function(child, parentClass) {
+    // initialize
+    child = child || {};
+    child.init = child.init || function() {};
+
+    function CreatedClass(...params) {
+      if (parentClass) {
+        parentClass.apply(this, params);
+      }
+      child.init.apply(this, params);
```

Comment:
개인적으로.. 여기도 존재하면 호출하는 것으로 처리하는게 좋을 것 같습니다.

자바스크립트 엔진이 `function() {}`코드를 만나면 어찌되었건 함수 객체가 하나는 생성되고, 메모리를 조금이라도 점유하게 되니까요 ㅎㅎ

Reply:
```js
let CreatedClass;
if (parentClass) {
  CreatedClass = function(...params) {
    parentClass.apply(this, params.slice(0, parentClass.length));
    child.init.apply(this, params.slice(parentClass.length));
  };
  CreatedClass.prototype = Object.create(parentClass.prototype);
  CreatedClass.prototype.constructor = CreatedClass;
} else {
  CreatedClass = function(...params) {
    child.init.apply(this, params);
  };
}
```
이런식으로 바꿔보았습니다!

Reply:
음 제 말은 그걸 말한게 아니었어요 ㅎㅎ

body가 빈 함수를 매번 생성해서 초기 `child.init` 에 optional 처리하지 않고 실행 자체를 안하면 될 것 같아서요.

```suggestion
      if (child.init && typof child.init === 'function') {
            child.init.apply(this, params);
      }
```
이렇게 되어야 할 것 같아서요 ㅎㅎ

Reply:
인수를 초기화 할 생각만 했지 위 방법으로는 생각하지 못했네요.

> `function() {}코드를 만나면 어찌되었건 함수 객체가 하나는 생성`

이제 이해가 된 것 같습니다! 수정했습니다!